src/smt/translate_exprs.rs

```diff
@@ -175,18 +207,24 @@ impl<'smt, 'ctx> TranslateExprs<'smt, 'ctx> {
                 panic!("illegal cast to {:?} from {:?}", &expr.ty, &operand.ty)
             }
             ExprKind::Quant(quant_op, quant_vars, ann, operand) => {
+                self.push(); // avoid leaking any fresh quant_vars into the outside scope
```

Comment:
Why is this now needed? Why wasn't this an issue before?

Reply:
I relied heavily on the functionality of the `SmtScope` for creating fresh quantified variables (creating the quantified fuel on demand).
The quantifier translation does actually not use this feature. Instead, a scope with all quantified variables is manually created.
The problem is that `mk_scope` calls `get_local` which calls `init_local` if the variable is new (this is always the case for `mk_scope`) and
`init_local` automatically adds the new variable to the current scope. Therefore, we have the variables twice. Once in the current scope and once in the scope created by `mk_scope`.

This is my easy "fix". Maybe this is actually a problem with `mk_scope` or with my understanding of the translation code.

Reply:
The explanation does make some sense to me. But I just do not understand why this issue wouldn't have shown up before already. Wouldn't any call to `mk_scope` always create the same variables on the outer quantifier as well? 

Could you maybe create a quick example HeyVL file with (nested?) quantifiers where this problem does/does not show up in the SMT output?

Also if this change is correct, definitely don't forget the case for `t_eureal` :)